### PR TITLE
Paginate project history

### PR DESCRIPTION
**Pull request by roberson-io**
This paginates project history (#5031 ).  I moved the pagination macro from the search results template to a separate template that the project history page also uses.

### CODE REVIEW

File: warehouse/templates/search/results.html [hunk 1]
```diff
@@ -44,50 +46,9 @@ <h3 class="package-snippet__title">
...
-    <a href="{{ map.next_page.href }}" class="button button-group__button">Next</a>
-    {% endif %}
-  </div>
-{% endif %}
-{%- endmacro %}
-
 {% block search_form_extra_inputs -%}
   {%- for classifier in applied_filters %}
-    <input name="c" type="hidden" value="{{ classifier|e }}">
+  <input name="c" type="hidden" value="{{ classifier|e }}">
```
Comment (pradyunsg) [inline warehouse/templates/search/results.html]: An unnecessary change.

Comment (roberson-io) [reply warehouse/templates/search/results.html]: I thought so, too, but the linter failed on this.  I changed it back, and it looks like the Travis lint job fails, too.  Thanks for taking a look!

**Comment by yeraydiazdiaz** [comment]
Thank for your PR @roberson-io, here are two screenshots of the functionality for reviewer convenience:

screen shot 2018-11-12 at 14 17 03
screen shot 2018-11-12 at 14 18 22

File: warehouse/manage/views.py [hunk 2]
```diff
@@ -729,13 +731,25 @@ def delete_project_role(project, request):
     permission="manage:project",
 )
 def manage_project_history(project, request):
-    journals = (
+    try:
+        page_num = int(request.params.get("page", 1))
+    except ValueError:
+        raise HTTPBadRequest("'page' must be an integer.") from None
```
Comment (yeraydiazdiaz) [inline warehouse/manage/views.py]: I believe you don't need `from None` here

File: tests/unit/manage/test_views.py [hunk 3]
```diff
@@ -1657,3 +1657,26 @@ def test_get(self, db_request):
...
+
+        url_maker = pretend.stub()
+        url_maker_factory = pretend.call_recorder(lambda request: url_maker)
+        monkeypatch.setattr(views, "paginate_url_factory", url_maker_factory)
+
+        project = ProjectFactory.create()
+        with pytest.raises(HTTPBadRequest):
+            views.manage_project_history(project, db_request)
+
+        assert page_cls.calls == []
```
Comment (yeraydiazdiaz) [inline tests/unit/manage/test_views.py]: Would you mind adding two additional positive tests?

Ideally creating >25 JournalEntry objects and asserting the first 25 entries are returned when retrieving the first page and asserting that the remaining entry is returned when requesting the second page.

Or less entries if you decide to move the 25 to a setting

File: warehouse/manage/views.py [hunk 4]
```diff
@@ -729,13 +731,25 @@ def delete_project_role(project, request):
...
         .options(joinedload("submitted_by"))
         .filter(JournalEntry.name == project.name)
         .order_by(JournalEntry.submitted_date.desc(), JournalEntry.id.desc())
-        .all()
     )
+
+    journals = SQLAlchemyORMPage(
+        journals_query,
+        page=page_num,
+        items_per_page=25,
```
Comment (yeraydiazdiaz) [inline warehouse/manage/views.py]: It may be convenient for testing to have this as a setting, possibly opening the door for a nice dropdown in the pagination controls in the future

File: warehouse/manage/views.py (hunk 2 above)
Comment (roberson-io) [reply warehouse/manage/views.py]: I removed `from None` here.

File: tests/unit/manage/test_views.py (hunk 3 above)
Comment (roberson-io) [reply tests/unit/manage/test_views.py]: I added the two tests you suggested plus a test for the case where someone requests a page out of range.  For example, requesting page 3 when there are only 2 pages.

File: warehouse/manage/views.py (hunk 4 above)
Comment (roberson-io) [reply warehouse/manage/views.py]: I would be happy to make `items_per_page` a setting, but I think I need some guidance on how I should go about doing that.  Thanks for reviewing and for your feedback!

Comment (yeraydiazdiaz) [reply warehouse/manage/views.py]: On second thought it might work better as a request paramenter read by the view, but only makes sense if we were to implement the feature in the frontend so for now it's fine as it is. 👍

**Comment by ewdurbin** [comment]
thank you @roberson-io!
